src/openforms/appointments/utils.py

```diff
@@ -146,6 +146,20 @@ def book_appointment_for_submission(submission: Submission) -> None:
             start_time=start_at,
         )
         appointment_register_success(appointment_info, client)
+        if submission.previous_submission:
+            try:
+                # Delete previous appointment if one exists
+                client.delete_appointment(
+                    submission.previous_submission.appointment_info.appointment_id
+                )
+                submission.previous_submission.appointment_info.status = (
+                    AppointmentDetailsStatus.cancelled
+                )
+                submission.previous_submission.appointment_info.save(
+                    update_fields=["status"]
+                )
+            except (AppointmentInfo.DoesNotExist, AppointmentDeleteFailed):
+                pass
```

Comment:
it seems weird that we silently fail if appointment delete fails? This should at least be logged